TrackletAlgorithm/MatchProcessor.h

```diff
@@ -809,133 +1501,97 @@ void MatchProcessor(BXType bx,
       //memory the projection points to
       
       // number of bits used to distinguish the different modules in each layer/disk
-      auto nbits_all = LAYER < trklet::N_LAYER ? nbitsallstubs[LAYER] : nbitsallstubs[trklet::N_LAYER + DISK];
+      constexpr auto nbits_all = nbitsallstubs[LAYER];
       
       // number of bits used to distinguish between VMs within a module
-      auto nbits_vmme = LAYER < trklet::N_LAYER ? nbits_vmmeall[LAYER] : nbits_vmmeall[trklet::N_LAYER + DISK];
+      constexpr auto nbits_vmme = nbits_vmmeall[LAYER];
+      constexpr auto nvmbits_ = nbits_vmme + nbitsallstubs[LAYER];
+      constexpr auto nbins_vmme = 1 << nbits_vmme;
       
       // bits used for routing
       iphi = iphiproj.range(iphiproj.length()-nbits_all-1,iphiproj.length()-nbits_all-nbits_vmme);
       
-      typename VMProjection<VMPTYPE>::VMPFINEPHI finephi = iphiproj.range(iphiproj.length()-nbits_all-nbits_vmme-1,
-									  iphiproj.length()-nbits_all-nbits_vmme-3); 
+      constexpr int kNfineBits = 3;
+      typename VMProjection<VMPTYPE>::VMPFINEPHI finephi = (iphiproj >> (iphiproj.length() - (nvmbits_ + kNfineBits))) & ((1 << kNfineBits) - 1);
       
-      int nextrabits = 2;
-      int overlapbits = nbits_vmme + nbits_all + nextrabits;
+      constexpr int nextrabits = 2;
+      constexpr int overlapbits = nbits_vmme + nbits_all + nextrabits;
       
-      unsigned int extrabits = iphiproj.range(iphiproj.length() - overlapbits-1, iphiproj.length() - overlapbits - nextrabits);
+      unsigned int extrabits = (iphiproj >> (iphiproj.length() - overlapbits -  nextrabits)) & ((1<<nextrabits) - 1);
 
       unsigned int ivmPlus = iphi;
-	
-      //ap_int<2> shift = 0;
-      
-      //If the projection goes to the first (ivmMinus) phiProjBin is zero but
-      //if the projection goes to the second bin ivmMinus+1 phiProjBin is one
-      typename ProjectionRouterBuffer<BARREL, APTYPE>::PHIPROJBIN phiProjBin  = 0; 
 
+      ap_int<2> shift = 0;
```

Comment:
You have added this shift - why is it needed? (I did not try to track the code to figure this out..)

Reply:
This is leftover from the rebase. I've removed it.

Reply:
This feeds into the MEU. I believe it shows whether the disk is `+` or `-` z

Reply:
I don't think this has anything to do with +/- z? I had used this variable in the past to indicate which phi bin you are processing. But I realized that I had duplicate data and got rid of this. The variable is still there, but commented out in the earlier code so I'm curious why you have reintroduced it.

Reply:
I must have started working with it before you removed it, and assumed it was essential to the disks. My implementation of the MEU currently relies on it to compute the fine phi variable.
https://github.com/cms-L1TK/firmware-hls/blob/2662e01f7ecf90a69e6f98e303403cffa5f8c830/TrackletAlgorithm/MatchEngineUnit.h#L72
https://github.com/cms-L1TK/firmware-hls/blob/2662e01f7ecf90a69e6f98e303403cffa5f8c830/TrackletAlgorithm/MatchEngineUnit.h#L177-L182
If you want me to remove it, I can try to rework the logic.

Reply:
I was able to remove this in favor of `phiProjBin`